torch/nested/__init__.py

```diff
@@ -186,3 +187,63 @@ def nested_tensor(tensor_list, *, dtype=None, layout=None, device=None, requires
         return nt
     else:
         raise RuntimeError(f"Specified layout is unsupported for nested tensors: {layout}")
+
+
+def narrow(tensor: Tensor, dim: int, start: Union[int, Tensor], length: Union[int, Tensor], layout=torch.strided) -> Tensor:
+    r"""
+Constructs a nested tensor (which might be a view) from :attr:`tensor`, a strided tensor. This follows
+similar semantics to torch.Tensor.narrow, where in the :attr:`dim`-th dimension the new nested tensor
+(maybe view) shows only the elements in the interval `[start, start+length]`. As nested representations
```

Comment:
wording / formatting nit: rather than mentioning views in-line here, I think it'd be clearer to talk about view / copy behavior for each layout in a separate paragraph after this one